src/host/readDataCooked.cpp

```diff
@@ -797,453 +807,484 @@ void COOKED_READ_DATA::_transitionState(State state) noexcept
     _state = state;
 }
 
-// Draws the contents of _buffer onto the screen.
-//
-// By using _buffer._dirtyBeg to avoid redrawing the buffer unless needed, we turn the amortized
-// time complexity of _readCharInputLoop() from O(n^2) (n(n+1)/2 redraws) into O(n).
-// Pasting text would quickly turn into "accidentally quadratic" meme material otherwise.
-//
-// NOTE: Don't call _flushBuffer() after appending newlines to the buffer! See _handlePostCharInputLoop for more information.
-void COOKED_READ_DATA::_flushBuffer()
+til::point COOKED_READ_DATA::_getViewportCursorPosition() const noexcept

```

Comment:
📝: this thru `_setCursorPosition` are pretty verbatim moved from `BufferState` at the top of the file